src/main.ts

```diff
@@ -297,15 +308,68 @@ export default class MediaDbPlugin extends Plugin {
-		let fileMetadata = this.modelPropertyMapper.convertObject(mediaTypeModel.toMetaDataObject());
-		let fileContent = '';
-		const template = options.attachTemplate ? await this.mediaTypeManager.getTemplate(mediaTypeModel, this.app) : '';
+		let template = await this.mediaTypeManager.getTemplate(mediaTypeModel, this.app);
+
+		if (this.settings.useDefaultFrontMatter || !template) {
+			let fileMetadata = this.modelPropertyMapper.convertObject(mediaTypeModel.toMetaDataObject());
+			let fileContent = '';
+			template = options.attachTemplate ? template : '';
+
+			({ fileMetadata, fileContent } = await this.attachFile(fileMetadata, fileContent, options.attachFile));
+			({ fileMetadata, fileContent } = await this.attachTemplate(fileMetadata, fileContent, template));
+
+			fileContent = `---\n${this.settings.useCustomYamlStringifier ? YAMLConverter.toYaml(fileMetadata) : stringifyYaml(fileMetadata)}---\n` + fileContent;
+			return fileContent;
+		} else {
+			const parts = template.split('---');
+
+			if (parts.length < 3) {
+				throw new Error('Cannot find YAML front matter for template.');
+			}
+
+			let frontMatter = parseYaml(parts[1]);
+			let fileContent: string = parts[2];
+
+			// Updating a previous file
+			if (options.attachFile) {
+				const previousMetadata = this.app.metadataCache.getFileCache(options.attachFile).frontmatter;
 
-		({ fileMetadata, fileContent } = await this.attachFile(fileMetadata, fileContent, options.attachFile));
-		({ fileMetadata, fileContent } = await this.attachTemplate(fileMetadata, fileContent, template));
+				// Use contents (below front matter) from previous file
+				fileContent = await this.app.vault.read(options.attachFile);
+				const regExp = new RegExp(this.frontMatterRexExpPattern);
+				fileContent = fileContent.replace(regExp, '');
+				fileContent = fileContent.startsWith('\n') ? fileContent.substring(1) : fileContent;
 
-		fileContent = `---\n${this.settings.useCustomYamlStringifier ? YAMLConverter.toYaml(fileMetadata) : stringifyYaml(fileMetadata)}---\n` + fileContent;
-		return fileContent;
+				// Update updated front matter with entries from the old front matter, if it isn't defined in the new front matter
+				Object.keys(previousMetadata).forEach(key => {
+					const value = previousMetadata[key];
+
+					if (!frontMatter[key] && value) {
+						frontMatter[key] = value;
+					}
+				});
+			}
+
+			// Ensure that id, type, and dataSource are defined
+			if (!frontMatter.id) {
+				frontMatter.id = mediaTypeModel.id;
+			}
+
+			if (!frontMatter.type) {
+				frontMatter.type = mediaTypeModel.type;
+			}
+
+			if (!frontMatter.dataSource) {
+				frontMatter.dataSource = mediaTypeModel.dataSource;
+			}
+
+			// Only support stringifyYaml for templater plugin
+			fileContent = `---\n${stringifyYaml(frontMatter)}---\n${fileContent}`;
+
+			fileContent = executeInlineScriptsTemplates(mediaTypeModel, fileContent);
```

Comment:
The setting should also warn about running JavaScript on note creation. In general, this is a delicate topic since a compromised API could result in arbitrary JavaScript being run on the user's computer.